hooks/emitter/channel.go

```diff
@@ -1,30 +1,47 @@
 package emitter
 
 import (
-	sdk "github.com/cosmos/cosmos-sdk/types"
-
 	"github.com/bandprotocol/chain/hooks/common"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ibcxfertypes "github.com/cosmos/cosmos-sdk/x/ibc/applications/transfer/types"
 	"github.com/cosmos/cosmos-sdk/x/ibc/core/04-channel/types"
 	channeltypes "github.com/cosmos/cosmos-sdk/x/ibc/core/04-channel/types"
 
 	oracletypes "github.com/bandprotocol/chain/x/oracle/types"
 )
 
-// handleMsgRequestData implements emitter handler for MsgRequestData.
-func (h *Hook) handleMsgRecvPacket(
-	ctx sdk.Context, txHash []byte, msg *types.MsgRecvPacket, evMap common.EvMap, extra common.JsDict,
-) {
-	packet := common.JsDict{
-		"is_incoming":  true,
-		"block_height": ctx.BlockHeight(),
-		"src_channel":  msg.Packet.SourceChannel,
-		"src_port":     msg.Packet.SourcePort,
-		"sequence":     msg.Packet.Sequence,
-		"dst_channel":  msg.Packet.DestinationChannel,
-		"dst_port":     msg.Packet.DestinationPort,
+func (h *Hook) handleFungibleTokenPacket(
```

Comment:
change name to `extractFungibleTokenPacket(ctx, packetData []byte,evMap common.EvMap, packet)`